bin/server/cmd/root.go

```diff
@@ -327,13 +343,69 @@ func initBranding(ctx context.Context, settingService api.SettingService) error
 	return nil
 }
 
-// Run will run the main server.
-func (m *Main) Run(ctx context.Context) error {
-	ctx, cancel := context.WithCancel(ctx)
-	m.serverCancel = cancel
+func useEmbeddedDB() bool {
+	return len(pgURL) == 0
+}
+
+func (m *Main) newDB() (*store.DB, error) {
+	if useEmbeddedDB() {
+		return m.newLocalDB()
+	}
+	return m.newExternalDB()
+}
+
+func (m *Main) newExternalDB() (*store.DB, error) {
+	m.l.Info("Initializing external PostgreSQL connection...", zap.String("pgURL", pgURL))
```

Comment:
```suggestion
	m.l.Info("Establishing external PostgreSQL connection...", zap.String("pgURL", pgURL))
```

Reply:
If we want to print pgURL, we probably need to sanitize it as it contains the password

Reply:
I remove printing `pgURL` for now. It comes to my mind some brittle logics to do regexp matching and replace for an in-place sanitization, especially if there is syntax error. It isn't quite necessary to log `pgURL` here.

Reply:
Let me try to do it with `url.Redacted`, but it will only possible after parse succeeded.

Reply:
As 48232c2e5bd5a43eba4a9459a4999995562a9b80